### PR TITLE
[Draft] boards: riscv: longan_nano: add configuration for LCD

### PR DESCRIPTION
**Pull request by soburi**
Add configuration for built-in display.
LCD has a Sitronix ST7735S controller that is connected to the SPI bus.
Using it with the ST7735R driver.

This commit need to merge after #44755 and #44795

### CODE REVIEW

File: samples/drivers/display/boards/longan_nano.conf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Copyright (c) 2022 TOKITA Hiroshi <tokita.hiroshi@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_SPI=y
```
Comment (jfischer-no) [inline samples/drivers/display/boards/longan_nano.conf:4]: NACK for unnecessary overlay, canbe part of the board configuration, see https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/pinetime_devkit0/Kconfig.defconfig#L14 as example.

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: boards/riscv/longan_nano/Kconfig.defconfig [hunk 2]
```diff
@@ -8,4 +8,20 @@ config BOARD
 	default "longan_nano" if BOARD_LONGAN_NANO
 	default "longan_nano_lite" if BOARD_LONGAN_NANO_LITE
 
+# This definition automatically resolves the dependency to GPIO
+# from spi1/cs-gpios defined in the DTS file.
+config GPIO
+	bool
+	default y if $(dt_nodelabel_enabled,spi0) && $(dt_nodelabel_has_prop,spi0,cs-gpios)
```
Comment (gmarull) [inline boards/riscv/longan_nano/Kconfig.defconfig:15]: why is this required? please just add `CONFIG_GPIO=y` to board defconfig if not already there (gpio is considered a "basic" peripheral so it can be enabled by default).

File: boards/riscv/longan_nano/longan_nano-common.dtsi [hunk 3]
```diff
@@ -96,3 +97,47 @@
...
+&spi0 {
+	status = "okay";
+	pinctrl-0 = <&spi0_default>;
+	pinctrl-names = "default";
+
+	cs-gpios = <&gpiob 2 GPIO_ACTIVE_LOW>;
+
+	/* longan nano has LCD with st7735s controller.
+	 * It can use with st7735r driver.
+	 */
```
Comment (gmarull) [inline boards/riscv/longan_nano/longan_nano-common.dtsi:110]:
```suggestion
```

redundant comment

File: boards/riscv/longan_nano/longan_nano-common.dtsi [hunk 4]
```diff
@@ -96,3 +97,47 @@
...
+		compatible = "sitronix,st7735r";
+		reg = <0>;
+		status = "okay";
+		label = "LCD0";
+		reset-gpios = <&gpiob 1 GPIO_ACTIVE_LOW>;
+		cmd-data-gpios = <&gpiob 0 GPIO_ACTIVE_LOW>;
+		width = <160>;
+		height = <80>;
+		invert-color;
+		swap-color-order;
```
Comment (gmarull) [inline boards/riscv/longan_nano/longan_nano-common.dtsi:121]: this property is not defined in the binding

**Review by gmarull — CHANGES_REQUESTED** [review]
